### PR TITLE
#109 Add color picker when create new calendar

### PR DESCRIPTION
**Pull request by pbson**
color2

### CODE REVIEW

File: src/linagora.esn.calendar/app/components/calendar-color-picker/calendar-color-picker.js [hunk 1]
```diff
@@ -43,7 +43,11 @@
 
     function set() {
       if (self.selected) {
-        self.color = self.CAL_LIST_OF_COLORS[self.selected];
+        if (self.selected in self.CAL_LIST_OF_COLORS) {
+          self.color = self.CAL_LIST_OF_COLORS[self.selected];
+        } else {
+          self.color = self.selected;
+        }
```
Comment (tuanlc) [inline src/linagora.esn.calendar/app/components/calendar-color-picker/calendar-color-picker.js:50]:
```suggestion
        self.color = self.selected in self.CAL_LIST_OF_COLORS ? self.CAL_LIST_OF_COLORS[self.selected] : self.selected;
```

File: src/linagora.esn.calendar/app/components/calendar-color-picker/calendar-color-picker.less [hunk 2]
```diff
@@ -9,11 +9,44 @@
     margin: 3px;
     .all-centered;
   }
+  .square-box-color-picker{
+    margin: 3px;
+    .all-centered;
+    border: 2px dashed #B5B5B5;
```
Comment (tuanlc) [inline src/linagora.esn.calendar/app/components/calendar-color-picker/calendar-color-picker.less:15]:
```suggestion
   .all-centered;

    margin: 3px;
    border: 2px dashed #B5B5B5;
```

**Comment by tuanlc** [comment]
LGTM!

**Comment by huy-ta** [comment]
I think we can add some auto-contrast (CSS) mechanism to the "+" mark so that when the user chooses a bright color, the "+" will still be easily seen.

File: src/linagora.esn.calendar/app/components/calendar-color-picker/calendar-color-picker.js (hunk 1 above)
Comment (pbson) [reply src/linagora.esn.calendar/app/components/calendar-color-picker/calendar-color-picker.js]: FIXED

File: src/linagora.esn.calendar/app/components/calendar-color-picker/calendar-color-picker.less (hunk 2 above)
Comment (pbson) [reply src/linagora.esn.calendar/app/components/calendar-color-picker/calendar-color-picker.less]: FIXED

File: src/linagora.esn.calendar/app/components/calendar-color-picker/calendar-color-picker.less [hunk 3]
```diff
@@ -9,11 +9,44 @@
...
+  input[type="color"] {
+    -webkit-appearance: none;
+    border: none;
+  }
+  input[type="color"]::-webkit-color-swatch-wrapper {
+    padding: 0;
+  }
+  input[type="color"]::-webkit-color-swatch {
+    border: none;
+  }
```
Comment (huy-ta) [inline src/linagora.esn.calendar/app/components/calendar-color-picker/calendar-color-picker.less:44]: Will these pollute `input[type="color"]` everywhere? Shouldn't you put it in some local CSS scope?

File: src/linagora.esn.calendar/app/components/calendar-color-picker/calendar-color-picker.pug [hunk 4]
```diff
@@ -6,6 +6,14 @@
       .modal-body
         .square-box(ng-repeat="color in colorKeys", class="{{color}}", ng-class="{selected: isSelected(color)}", ng-click="select(color)")
           .color-box(ng-style="{'background-color': CAL_LIST_OF_COLORS[color]}")
+        .square-box-color-picker(ng-click="select(colorValue)", ng-class="{selected: isSelected(colorValue)}")
+          input.calendar-theme-color-picker(
+            type = "color",
+            ng-model = "colorValue",
+            ng-change = "select(colorValue)",
+            ng-init = "colorValue = color",
```
Comment (huy-ta) [inline src/linagora.esn.calendar/app/components/calendar-color-picker/calendar-color-picker.pug:14]:
```suggestion
            type="color",
            ng-model="colorValue",
            ng-change="select(colorValue)",
            ng-init="colorValue = color"
```

**Comment by huy-ta** [comment]
LGTM, make sure you do these things though:

- [ ] Write tests for the changes
- [ ] Test it in our supported browsers (Firefox, Chrome, Safari)
- [ ] Test it in mobile devices

File: src/linagora.esn.calendar/app/components/calendar-color-picker/calendar-color-picker.pug (hunk 4 above)
Comment (pbson) [reply src/linagora.esn.calendar/app/components/calendar-color-picker/calendar-color-picker.pug]: FIXED

File: src/linagora.esn.calendar/app/components/calendar-color-picker/calendar-color-picker.less [hunk 5]
```diff
@@ -9,11 +9,45 @@
...
+  
+  i.mdi.mdi-plus{
+    position: absolute;
+    pointer-events:none;
```
Comment (huy-ta) [inline src/linagora.esn.calendar/app/components/calendar-color-picker/calendar-color-picker.less:49]:
```suggestion
    pointer-events: none;
```

File: src/linagora.esn.calendar/app/components/calendar-color-picker/calendar-color-picker.less [hunk 6]
```diff
@@ -9,11 +9,45 @@
...
     width: 36px;
     height: 36px;
   }
   .selected {
     border: 1px solid @m-red;
   }
+  .calendar-theme-color-picker{
+    display: flex;
+    justify-content: center;
+    align-items: center;
```
Comment (huy-ta) [inline src/linagora.esn.calendar/app/components/calendar-color-picker/calendar-color-picker.less:27]:
```suggestion
    .all-centered;
```

**Review by huy-ta — APPROVED** [review]

File: src/linagora.esn.calendar/app/components/calendar-color-picker/calendar-color-picker.less [hunk 7]
```diff
@@ -9,11 +9,45 @@
...
+    width: 39px;
+    height: 39px;
+  }
+  .calendar-theme-color-picker:hover{
+    cursor: pointer;
+  }
```
Comment (huy-ta) [inline src/linagora.esn.calendar/app/components/calendar-color-picker/calendar-color-picker.less:33]:
```suggestion
    height: 39px;
    
    &:hover{
      cursor: pointer;
    }
  }
```

File: src/linagora.esn.calendar/app/components/calendar-color-picker/calendar-color-picker.less [hunk 8]
```diff
@@ -9,11 +9,45 @@
...
+    border: none;
+    padding: 0;
+  }
+  input[type="color"]::-webkit-color-swatch-wrapper {
+    padding: 0;
+  }
+  input[type="color"]::-webkit-color-swatch {
+    border: none;
+  }
+
```
Comment (huy-ta) [inline src/linagora.esn.calendar/app/components/calendar-color-picker/calendar-color-picker.less:46]:
```suggestion
    padding: 0;

    &::-webkit-color-swatch-wrapper {
      padding: 0;
    }

    &::-webkit-color-swatch {
      border: none;
    }
  }  
```

**Review by rezk2ll — APPROVED** [review]

**Review by huy-ta — CHANGES_REQUESTED** [review]
abc

**Review by huy-ta — APPROVED** [review]

**Review by tuanlc — APPROVED** [review]

**Pull request merged**